latexify/core.py

```diff
@@ -163,9 +167,44 @@ def visit_Compare(self, node):
 
     if isinstance(node.ops[0], ast.Eq):
       return lstr + '=' + rstr
+    # CHANGE:(20200801@1MLightyears) Add support for conditions for <,>,\ge,\le,\ne,is
+    elif isinstance(node.ops[0], ast.Gt):
+      return lstr + '>' + rstr
+    elif isinstance(node.ops[0], ast.Lt):
+      return lstr + '<' + rstr
+    elif isinstance(node.ops[0], ast.GtE):
+      return lstr + r'\ge ' + rstr
+    elif isinstance(node.ops[0], ast.LtE):
+      return lstr + r'\le ' + rstr
+    elif isinstance(node.ops[0], ast.NotEq):
+      return lstr + r'\ne ' + rstr
+    elif isinstance(node.ops[0], ast.Is):
+      return lstr + r'\space is\space ' + rstr
+    elif isinstance(node.ops[0], ast.IsNot):
+      return lstr + r'\space is\space not\space ' + rstr
+
     else:
       return r'\operatorname{unknown\_comparator}(' + lstr + ', ' + rstr + ')'
 
+  # CHANGE:(20200801@1MLightyears) Add support for BoolOp, which allow Latexify to
+  # deal with more complex boolean statements.
+  # The structure is weak, for I seldom use very complex boolean statements and
+  # currently have no idea how to simply implement a result like "x=0,1" instead of
+  # "x=0 Or x=1" as it is now XD
+  def visit_BoolOp(self, node):
```

Comment:
This function needs the same strategy to `visit_BinOp` to deal with parenthesis appropriately (generally, logical operators has priorities: *not* > *and* > *or*)

Reply:
for "visit_BoolOp"
At the first version of `visit_BoolOp`, I actually wrote the `not` parts. But after testing a few cases I found it kind of silly to have output like "$if not x\equal 10$", it's not something would appear in anyone's paper. Instead what we actually want is "$if x\neq 10$", but that needs a new list or dict to detect and find proper negative statements. I will work on that after next commit.
About the parenthesis, it is true that parenthesis needs to be mentioned, but when it comes to LaTeX output it needs to be reader-friendly. One approach is to add brackets, another is to start a new line when indicating priority like:
```
if x < 1 \lor x > 10\\
\land x>0
```
for `if (x<1 or x>10)and(x>0)`. 
"$\lor$" and "$\land$" seem not so good at such case btw.

Reply:
As far as the papers I read, most authors tend to use comma to connect statements have AND relationship, because OR relationship is quite rarely seen. ORs are likely to be clearly indicated when needed.
If you have a better output format for this plz reply me, and I will try to implement the brackets for parenthesis in next commit.

Reply:
I prefer single-line form with `\neg`, `\lor` and `\land`. It keeps the same syntax to Python except appearances of operators, and would eventually not surprise users. I think we have to keep minimality about extra conventions of LaTeX since any styles are basically not a common consensus of us (though there are already some strong assumptions introduced when I wrote this first).

Reply:
I agree with the keeping minimality about extra conventions. I will try to implement the brackets for parenthesis in single-line form in the next commit.